internal/pkg/cli/job_deploy.go

```diff
@@ -300,10 +303,15 @@ func (o *deployJobOpts) stackConfiguration(addonsURL string) (cloudformation.Sta
 }
 
 func (o *deployJobOpts) runtimeConfig(addonsURL string) (*stack.RuntimeConfig, error) {
+	svcDiscovery, err := envUsesLegacySvcDiscovery(o.envCFN, o.appName, o.envName)
+	if err != nil {
+		return nil, err
+	}
 	if !o.buildRequired {
 		return &stack.RuntimeConfig{
-			AddonsTemplateURL: addonsURL,
-			AdditionalTags:    tags.Merge(o.targetApp.Tags, o.resourceTags),
+			AddonsTemplateURL:      addonsURL,
+			AdditionalTags:         tags.Merge(o.targetApp.Tags, o.resourceTags),
+			LegacyServiceDiscovery: svcDiscovery,
```

Comment:
does copilot job use service discovery 🤔

Reply:
We inject the endpoint as an environment variable, so we still have to update this here.